### PR TITLE
Feature/cypress 4

### PR DESCRIPTION
**Pull request by hayleyw7**
# Pull Request

## What was implemented, refactored, or otherwise changed?

- Make commands file more dynamic & more dry
- Add more sad path tests

## Issues, comments, questions, or concerns (if any)

- 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor
- [ ] Setup

## How tested
- [ ] Browser
- [ ] console.log()
- [ ] Dev tools
- [x] Cypress

## Checklist
- [x] Style guidelines
- [x] Self-review
- [x] Commented code if necessary

### CODE REVIEW

File: cypress/integration/all_restrooms_spec.js [hunk 1]
```diff
@@ -11,41 +11,28 @@ describe('Restrooms Page - All', () => {
...
-    cy.wait(1000)
-    .get('h3[alt="business"]')
+    cy.get('h3[alt="business"]')
       .contains('Donald Duck Den')
     .get('p[alt="city and state"]')
       .contains('Toledo, OH')
     .get('p[alt="miles away"]')
       .contains('0.61 mi')
-  })  
+  })
```
Comment (sarahrudy) [inline cypress/integration/all_restrooms_spec.js:23]: I like how you're targeting the h3 tag with the alt 👍

File: cypress/integration/gender_free_restrooms.js [hunk 2]
```diff
@@ -4,36 +4,15 @@ describe('Restrooms Page - Gender Free', () => {
...
-  //     .should('not.contain', 'Center For Performing Arts')
-  // }) 
-
-  // uncomment this once stub works
-
   it('should not display gendered restrooms', () => {
     // cy.wait(1000)
     cy.get('h3[alt="business"]')
       .should('not.contain', 'Donald Duck Den')
   })
```
Comment (sarahrudy) [inline cypress/integration/gender_free_restrooms.js:11]: Nice!

File: cypress/integration/search_page_spec.js [hunk 3]
```diff
@@ -37,13 +36,61 @@ describe('Search Page', () => {
...
+    .get('h3[class="bad-zip"]') 
+      .contains('Please enter a valid US zip code.')
+  });
+
   it('should go to restrooms page when search button clicked', () => {
     cy.get('input[alt="Enter Zip Code"]')
       .type('43606')
+    .get('button[alt="Show List Button"]')
+      .click()
+  })
```
Comment (sarahrudy) [inline cypress/integration/search_page_spec.js:73]: Really thorough testing here -- great job!

**Review by sarahrudy — APPROVED** [review]
You're crushin' cypress testing, Hayley!

**Comment by sarahrudy** [comment]
You're crushin' cypress testing, Hayley! Awesome job. 🙌

File: cypress/integration/all_restrooms_spec.js (hunk 1 above)
Comment (hayleyw7) [reply cypress/integration/all_restrooms_spec.js:23]: Thanks!

File: cypress/integration/search_page_spec.js (hunk 3 above)
Comment (hayleyw7) [reply cypress/integration/search_page_spec.js:73]: Thanks!

**Pull request merged**